ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Update.hs

```diff
@@ -33,17 +36,72 @@ import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog
 import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog.Update
 import           Ouroboros.Consensus.Storage.LedgerDB.ReadsKeySets
 import           Ouroboros.Consensus.Util.IOLike
+import qualified Ouroboros.Network.AnchoredSeq as AS
+
+data LedgerDBUpdate = Prune | Extend
+  deriving (Show)
 
 -- | PRECONDITION: The new 'LedgerDB' must be the result of calling either
 -- 'LedgerDB.ledgerDbSwitch' or 'LedgerDB.ledgerDbPushMany' on the current
 -- 'LedgerDB'.
 setCurrent ::
-     MonadSTM m
+     forall blk m.
+     ( MonadSTM m
+     , IsLedger (LedgerState blk)
+     , HasLedgerTables (LedgerState blk)
+     , HasCallStack
+     )
   => StrictTVar m (DbChangelog' blk)
   -> DbChangelog' blk
+  -> LedgerDBUpdate
   -> STM m ()
-setCurrent = writeTVar
+setCurrent v dblog = \case
+  Prune ->
+    updatePruning (changelogLastFlushedState dblog)
+  Extend ->
+    updateExtending (changelogVolatileStates dblog) (changelogDiffs dblog)
+ where
+   updatePruning :: ExtLedgerState blk EmptyMK -> STM m ()
+   updatePruning lf =
+     let
+       f :: (Ord k, Eq v) => SeqDiffMK k v -> SeqDiffMK k v
```

Comment:
I wonder if we could give `f` a more descriptive name or comment what's happening here.